q2_feature_table/_summarize/summarize_assets/src/chart.js

```diff
@@ -0,0 +1,154 @@
+// ----------------------------------------------------------------------------
+// Copyright (c) 2016-2017, QIIME 2 development team.
+//
+// Distributed under the terms of the Modified BSD License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+// ----------------------------------------------------------------------------
+
+import naturalSort from 'javascript-natural-sort';
+import * as d3 from 'd3';
+
+
+const categoricalBin = (data, counts, x) => {
+  const result = [];
+  x.domain().forEach((cat) => {
+    const catArray = Object.keys(data)
+                      .filter(sampleID => data[sampleID] === cat)
+                      .map(sampleID => counts[sampleID]);
+    catArray.x0 = x(cat);
+    result.push(catArray);
+  });
```

Comment:
Now that I look at it, is it weird to be using this `x.domain().forEach()` method, instead of a `for (const cat of x.domain()) {}`? Same thing, mainly just readability maybe?